src/rules/use_isnan.rs

```diff
@@ -49,6 +49,57 @@ impl LintRule for UseIsNaN {
       ProgramRef::Script(ref s) => visitor.visit_script(s, &DUMMY_NODE),
     }
   }
+
+  fn docs(&self) -> &'static str {
+    r#"Disallows comparisons to `NaN`.
+
+Because `NaN` is unique in JavaScript by not being equal to anything, including itself, the results of comparisons to `NaN` are confusing:
+
+- `NaN === NaN` or `NaN == NaN` evaluate to false
+- `NaN !== NaN` or `NaN != NaN` evaluate to true
+
+Therefore, this rule makes you use the `isNaN()` or `Number.isNaN()` to judge the value is `NaN` or not.
+
+### Invalid:
+```typescript
+if (foo == NaN) {
+    // ...
+}
+```
+
+```typescript
+if (foo != NaN) {
+    // ...
+}
+```
+
+```typescript
+switch (foo) {
+  case NaN:
+    // ...
+}
+```
+
+### Valid:
+```typescript
+if (isNaN(foo)) {
+    // ...
+}
+```
+
+```typescript
+if (!isNaN(foo)) {
+    // ...
+}
+```
+
+```typescript
+switch (foo) {
+  case isNaN(foo):
+    // ...
+}
```

Comment:
This looks a little bit strange in semantics, because `isNaN(foo)` evaluates to boolean but `foo` is number, thus `foo` is never equal to  `isNaN(foo)`.
I think we need to use `if` instead of `switch` in order to convert the `case NaN:` example into a valid one. So we could probably remove this example.

Instead, how about adding the following to invalid examples? It is also invalid and should be converted into `if` statement, so it would be worth presenting to readers.

```ts
switch (NaN) {
  case foo:
    // ...
}
```

Reply:
Definitely. I misunderstood.
I'll remove  previous changes and add `switch(NaN)`  `case NaN`  as invalid example.